controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -120,134 +160,384 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileCephFilesystemSubVolumeGroup(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileNoobaaAccount(); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.consolidateStatus(); err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateDeleting
 	}
 
-	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateConfiguring
-	r.storageConsumer.Status.CephResources = []*ocsv1alpha1.CephResourcesSpec{}
+	return reconcile.Result{}, nil
+}
 
-	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
+func (r *StorageConsumerReconciler) reconcileCephRadosNamespace() error {
+	blockPools := &rookCephv1.CephBlockPoolList{}
+	if err := r.List(r.ctx, blockPools, client.InNamespace(r.namespace)); err != nil {
+		return err
+	}
 
-		// A provider cluster already has a NooBaa system and does not require a NooBaa account
-		// to connect to a remote cluster, unlike client clusters.
-		// A NooBaa account only needs to be created if the storage consumer is for a client cluster.
-		clusterID := util.GetClusterID(r.ctx, r.Client, &r.Log)
-		if clusterID != "" && !strings.Contains(r.storageConsumer.Name, clusterID) {
-			if err := r.reconcileNoobaaAccount(); err != nil {
-				return reconcile.Result{}, err
+	// ensure for this consumer a rados namespace is created in every blockpool
+	for idx := range blockPools.Items {
+		bp := &blockPools.Items[idx]
+		if bp.Name == "builtin-mgr" {
+			continue
+		}
+
+		rns := &rookCephv1.CephBlockPoolRadosNamespace{}
+		rns.Name = func() string {
+			if r.isUpdatedProvider && strings.HasPrefix(bp.Name, r.storageCluster.Name) {
+				return r.radosnamespaceName
+			}
+			return fmt.Sprintf("%s-%s", bp.Name, r.radosnamespaceName)
+		}()
+
+		rns.Namespace = r.namespace
+
+		if _, err := ctrl.CreateOrUpdate(r.ctx, r.Client, rns, func() error {
+			// if rns was created in provider mode storagerequest will be the controller owner
+			// as only single controller owner is permitted remove the old and place consumer as the owner
+			if existing := metav1.GetControllerOfNoCopy(rns); existing != nil &&
+				existing.Kind != r.storageConsumer.Kind {
+				existing.BlockOwnerDeletion = nil
+				existing.Controller = nil
 			}
+			if err := r.own(rns); err != nil {
+				return err
+			}
+			rns.Spec.Name = r.radosnamespaceName
+			return nil
+		}); err != nil {
+			return err
 		}
 
-		cephResourcesReady := true
-		for _, cephResource := range r.storageConsumer.Status.CephResources {
-			if cephResource.Phase != "Ready" {
-				cephResourcesReady = false
-				break
+		phase := ""
+		if rns.Status != nil {
+			phase = string(rns.Status.Phase)
+		}
+		r.setCephResourceStatus(rns.Name, "CephBlockPoolRadosNamespace", phase)
+
+	}
+
+	//TODO: set this properly
+	osdCaps := fmt.Sprintf("profile rbd namespace=%s", r.radosnamespaceName)
```

Comment:
<implicit> should be replaced with empty string, maybe creation of cephclients could be part of another PR or let's meet if you want that to be included in this PR itself.

Reply:
I agree, moving the cephClients into it's own PR